lib/src/util/leak_detector_visitor.dart

```diff
@@ -147,15 +146,15 @@ abstract class LeakDetectorVisitor extends SimpleAstVisitor {
 
   @override
   void visitFieldDeclaration(FieldDeclaration node) {
-    ClassDeclaration classDecl = node.getAncestor((a) => a is ClassDeclaration);
+    CompilationUnit unit = node.getAncestor((a) => a is CompilationUnit);
     node.fields.variables.forEach(_buildVariableReporter(
-        classDecl, _fieldPredicateBuilders, rule, predicates));
+        unit, _fieldPredicateBuilders, rule, predicates));
   }
 
   @override
   void visitVariableDeclarationStatement(VariableDeclarationStatement node) {
-    FunctionBody function = node.getAncestor((a) => a is FunctionBody);
+    CompilationUnit unit = node.getAncestor((a) => a is CompilationUnit);
```

Comment:
Reverted, I was trying to refactor both methods but I couldn't since the callback is different and didn't completely reverted back.